### ISSUE TITLE
Not Saving

### ISSUE THREAD

**Issue by CatsLover2006** [Brionjv/3ds-Nintendogs-cats-Save-Editor#1]
Details:
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.InvalidOperationException: Form that is already displayed modally cannot be displayed as a modal dialog box. Close the form before calling showDialog.
   at System.Windows.Forms.Form.ShowDialog(IWin32Window owner)
   at System.Windows.Forms.Form.ShowDialog()
   at Nintendogs___Cats_Save_Editor.Form1.Writefile() in C:\Users\Brion\Documents\Visual Studio 2012\Projects\Nintendogs + Cats Save Editor\Nintendogs + Cats Save Editor\Form1.vb:line 99
   at Nintendogs___Cats_Save_Editor.Form1.Button2_Click(Object sender, EventArgs e) in C:\Users\Brion\Documents\Visual Studio 2012\Projects\Nintendogs + Cats Save Editor\Nintendogs + Cats Save Editor\Form1.vb:line 57
   at System.Windows.Forms.Control.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ButtonBase.WndProc(Message& m)
   at System.Windows.Forms.Button.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)

************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4018.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework/v4.0.30319/mscorlib.dll
----------------------------------------
Nintendogs + Cats Save Editor
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///D:/Nintendogs%20+%20cats%20save%20editor/Nintendogs%20+%20Cats%20Save%20Editor.exe
----------------------------------------
Microsoft.VisualBasic
    Assembly Version: 10.0.0.0
    Win32 Version: 14.8.3761.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/Microsoft.VisualBasic/v4.0_10.0.0.0__b03f5f7f11d50a3a/Microsoft.VisualBasic.dll
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3928.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4001.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3761.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3761.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4018.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3761.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
System.Runtime.Remoting
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3761.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Runtime.Remoting/v4.0_4.0.0.0__b77a5c561934e089/System.Runtime.Remoting.dll
----------------------------------------
PackageIO
    Assembly Version: 0.1.7.0
    Win32 Version: 0.1.7.0
    CodeBase: file:///D:/Nintendogs%20+%20cats%20save%20editor/PackageIO.DLL
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.

**Comment by Brionjv** [comment]
Make sure you have all files in same save editor folder :
- Microsoft.VisualBasic.PowerPacks.Vs.dll
- Nintendogs + Cats Save Editor.exe
- Nintendogs + Cats Save Editor.pdb
- Nintendogs + Cats Save Editor.xml
- PackageIO.dll

**Comment by CatsLover2006** [comment]
Checked for it. The files were there.

**Comment by Brionjv** [comment]
I don't know what happen. Try to delete temporary files of this application. It's probably a wrong configuration on your laptop.

**Comment by feroxi-s** [comment]
Hi,
Found this error that hadn't been fixed so I decided to try and fix it myself. Note that I don't actually know Visual Basic, but I did some Googling and tinkering and this is what I found to have worked.
Navigate to and open `Form1.vb` in the source code. Find and replace all occurrences of `Show()` with `ShowDialog()`. I'm not actually sure if every instance needs to be changed for it to work, but I haven't put in the time to figure out exactly which ones need changing.
[Here's the file](https://github.com/Brionjv/3ds-Nintendogs-cats-Save-Editor/files/4061811/Nintendogs.cats.save.editor1.0.1_fix.zip) if you don't feel like going in and doing it yourself. Maybe one day I'll actually learn to fork this properly, but this is what I have for now 😃
